network/agent/agent.ts

```diff
@@ -2,6 +2,7 @@ import { URL } from 'url'
 import HttpAgent from 'agentkeepalive'
 import LRU from 'lru-cache'
 import { getProxyAgent, ProxyAgentOptions } from '@pnpm/network.proxy-agent'
+import { getMaxParts, getFromUri } from '@pnpm/network.url';
```

Comment:
don't export getMaxParts. That can be an internal function in the network.url component

Reply:
also, the name of this component is probably not the best. A better one would be `@pnpm/network.config`

the function can be `pickSettingByUrl`

Reply:
But it takes an array of strings as argument, shouldn't it be an internal of `network.agent`? So I can pass the list of client certificates.

Reply:
Ah ok, nevermind